### PR TITLE
Association

### PR DESCRIPTION
**Pull request by danielsafari143**
This pull request adds new classes Classroom, Book, and Rental to the school library management system. The classes are defined as follows:

- Classroom: A classroom has a label and can have many students.
- Book: A book has a title and author.
- Rental: A rental has a date and is associated with a book and a person.

The changes also include the following:

- A has-many/belongs-to relationship between Classroom and Student.
- A has-many-through relationship between Person and Book using the intermediate class Rental.

Please review the changes and let me know if you have any questions.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 require_relative 'person'
 
 class Student < Person
+  attr_accessor :classroom
```
Comment (Davidosky007) [inline student.rb:5]: * Kindly add a method that serves as the 'belongs-to' side (a student belongs to a classroom) Link to the article with examples: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships  
*  [Check requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md#:~:text=Create%20the%20belongs%2Dto%20side%20(a%20student%20belongs%20to%20a%20classroom).)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @danielsafari143 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Ruby syntax written nicely 👍 
* Well written PR 👍 
* No linter errors 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 2]
```diff
@@ -11,8 +12,16 @@ def initialize(age, classroom, name, parent_permission)
   def play_hooky
     '¯(ツ)/¯'
   end
+
+  def add_classroom(classroom)
+    @classroom = classroom
+  end
```
Comment (codepantha) [inline student.rb]: - [ ] You are on the right track to correctly implement the `belongs-to` side. All that is left is to push the student into the `student` array in classroom as long as the student doesn't already exist in the classroom. See hint below 👇 
```ruby
def add_classroom=(classroom)
      @classroom = classroom
      classroom.student.push(self) unless classroom.student.include?(self)
  end
```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @danielsafari143 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Created associations between classes 📌 
✔️ Descriptive PR
✔️ No linter errors



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @danielsafari143 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Nice work on adding some tests in classes 💯 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**